bastionzero/target/attributes.go

```diff
@@ -233,3 +234,46 @@ func FlattenControlChannelSummary(ctx context.Context, apiObject *targets.Contro
 		})
 	}
 }
+
+type DatabaseAuthenticationConfig struct {
+	AuthenticationType   types.String `tfsdk:"authentication_type"`
+	CloudServiceProvider types.String `tfsdk:"cloud_service_provider"`
+	Database             types.String `tfsdk:"database"`
+	Label                types.String `tfsdk:"label"`
+}
+
+func DatabaseAuthenticationConfigAttribute() schema.Attribute {
+	return schema.SingleNestedAttribute{
+		Computed:    true,
+		Description: "Information about the db target's database authentication configuration.",
+		Attributes: map[string]schema.Attribute{
+			"authentication_type": schema.StringAttribute{
+				Computed:    true,
+				Description: "The type of authentication used when connecting to the database.",
+			},
+			"cloud_service_provider": schema.StringAttribute{
+				Computed:    true,
+				Description: "Cloud service provider hosting the database. Only used for certain types of authentication, such as ServiceAccountInjection.",
+			},
+			"database": schema.StringAttribute{
+				Computed:    true,
+				Description: "The type of database running on the target.",
+			},
+			"label": schema.StringAttribute{
+				Computed:    true,
+				Description: "User-friendly label for this database authentication configuration.",
+			},
+		},
+	}
+}
+
+func FlattenDatabaseAuthenticationConfig(ctx context.Context, apiObject dbauthconfig.DatabaseAuthenticationConfig) types.Object {
+	attributeTypes, _ := internal.AttributeTypes[ControlChannelSummaryModel](ctx)
+
+	return types.ObjectValueMust(attributeTypes, map[string]attr.Value{
+		"authentication_type":    types.StringPointerValue(apiObject.AuthenticationType),
+		"cloud_service_provider": types.StringPointerValue(apiObject.CloudServiceProvider),
+		"database":               types.StringPointerValue(apiObject.Database),
+		"label":                  types.StringPointerValue(apiObject.Label),
+	})
+}
```

Comment:
All of this code is highly specific to `dbtarget`, so lets move it to `dbtarget/dbtarget.go`. We can move it back to `target/attributes.go` if other target-related data sources need it.

I think even if we plan on adding a data source to fetch supported database auth configs, it still probably makes sense to keep it in `dbtarget.go` (and the package `dbtarget`) because most likely that data source will exist in the `dbtarget` package as well.

Reply:
Will do. I almost did that but thought maybe the established pattern was to dump such things here.

Reply:
Yes. Thanks for bringing this up--I'll use this thread to put some of my thoughts behind the structure.

So typically what I've been doing is putting common attributes used by all resources / data sources in the top package (`targets` in this case) in `attributes.go`, so that's why `ControlChannelSummaryModel` is there as its used by two data sources `bzerotarget` package and `clustertarget` package (which are sub-folders in `targets`).

But for specific things that are only used by one package, I try to keep the model structs as close as possible to where they're used. An example of this would be this one in this PR (`DatabaseAuthenticationConfig` model) or for example `ChildPolicyModel` in `jit.go` / package `jit` (instead of `policy/attributes.go` / package `policy`) because child policy is specific to jit.

I really should make a style guide for this repo to help solidify these style guidelines.